File: convec.cpp
#include "convec.h"
#include "convec_glf.h"
#include "convec_pvm.h"
#include "test_cases.h"

void convec(Matrix<double>& u, double h, Matrix<double>& K)
{
	struct test_cases* pt_test = get_tests();
	
	if (pt_test->convec_type == 0) {
		convec_pvm(u, h, K);
	} else {
		convec_glf(u, h, K);
	} 
}

File: convec.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
function K=convec(u, h, K)
%int:convec_type
%double:h
%matrix:u,K

global convec_type;

if ( convec_type == 0 )
  K=convec_pvm(u, h, K);
else
  K=convec_glf(u, h, K);
end